### ISSUE TITLE
MyBB admin home and logs page fails to load

### ISSUE THREAD

**Issue by hatchetboy** [jamiesage123/Project-HoneyPot-for-MyBB#2]
When this plugin is activated, the MyBB admin page and log pages fail to load with the dreaded error 500 screen.  When I deactivate it they load ok.  I can't however see the logs to find out what is going wrong!

**Comment by jamiesage123** [comment]
Hey, HatchetBoy!

Thanks for reporting this issue.

On a clean installation, this plugin seems to work fine!

Could you give me your MyBB version, PHP and MySQL version (if possible!)  and a list of your plugins that you have installed that may be conflicting with this one.

It is quite hard to debug this issue without any further information.

Cheers,

Jamie

**Comment by hatchetboy** [comment]
Yes, sorry - I thought it was a pretty useless bug report when I submitted it, but thought it was best to give you what I had!

I have:

MyBB

MyBB Version | 1.8.12
PHP Version | 5.6.31
SQL Engine | MySQLi 5.6.35

Active Plugins
reCAPTCHA Plugin (1.4)
Registration Security Question (1.2)
Tapatalk (4.5.6)

Inactive Plugins
Hello World! (2.0)
Project HoneyPot for MyBB (0.2)
Stop forum spam (1.2)

Literally works fine if Project HoneyPot is inactive, but if I activate it the "Tools and Maintence" and "Home" tabs in the admin console go to 500 status.  Means that I can't get to the log file. (If you can give me some advice where that log file is stored I might be able to get to it.  It's a host server so I can't get console access, but I do have file access).

**Comment by jamiesage123** [comment]
You're absolutely right! I'd rather people report a problem with less information than them not reporting it at all.

I really appreciate you responding with your set up information.

I did attempt to have a look for the reCAPTCHA and Registration Security Question plugins but I could only find ones which were not supported in MyBB 1.8.12.

However, I did try this plugin with Tapatalk just to verify that they are compatible together and seems so.

If you can attempt to figure out what the actual error is when you retrieve the 500 page, it would be very helpful.

You will need to find your PHP error log file which will hopefully contain the error log entry.

I am not sure on what type of access you have to your site. Error logs location totally depend on your site setup.

For example, if you're running Nginx, sometimes you can find your logs here:
- /var/log/nginx/error.log

Or if you're running apache, it can be here:
- /var/log/apache2/error.log

I will have a fiddle about and see if I can break this plugin and replicate your issue but I have had no luck so far.

**Comment by hatchetboy** [comment]
Helpfully, it's hosted on a rubbish shared hosting provider, so I don't have var access.  I'll raise a support ticket and see if they can extract the logs for me!

**Comment by hatchetboy** [comment]
Got it!

16-Aug-2017 08:55:58 UTC] PHP Fatal error:  Call to undefined function honeypot_admin_tools_permissions() in /home/<user>/public_html/forum/inc/class_plugins.php on line 139

**Comment by jamiesage123** [comment]
Amazing, thank you very much for finding that.

Looks like I forgot to remove an unused hook.

I shall push a fix for this once I get back from work.

Cheers for that!

**Comment by jamiesage123** [comment]
This has been fixed in v0.2.1

https://github.com/jamiesage123/Project-HoneyPot-for-MyBB/commit/de67c7f353ff1cd948fda6ecfad7ad1a9a3e7543

https://github.com/jamiesage123/Project-HoneyPot-for-MyBB/releases/tag/0.2.1

I have also updated the plugin on Extend MyBB. This should be updated once a mybb staff member have approved the build.

Thanks for reporting this issue and providing all the additional information!
